egs/wsj/s5/steps/nnet3/multilingual/allocate_multilingual_examples.py

```diff
@@ -0,0 +1,310 @@
+import libs.common as common_lib
+
+logger = logging.getLogger('libs')
+logger.setLevel(logging.INFO)
+handler = logging.StreamHandler()
+handler.setLevel(logging.INFO)
+formatter = logging.Formatter("%(asctime)s [%(pathname)s:%(lineno)s - "
+                              "%(funcName)s - %(levelname)s ] %(message)s")
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+logger.info('Starts generating multilinual examples (allocate_multilingual_examples.py)')
+
+def get_args():
+
+    parser = argparse.ArgumentParser(description="Writes ranges.*, outputs.* and weights.* files "
+                                   "in preparation for dumping egs for multilingual training.",
+                                   epilog="Called by steps/nnet3/multilingual/get_egs.sh")
+    parser.add_argument("--samples-per-iter", type=int, default=40000,
+                        help="The target number of egs in each archive of egs, "
+                        "(prior to merging egs). ")
+    parser.add_argument("--num-jobs", type=int, default=20,
+                        help="This can be used for better randomness in distributing "
+                        "languages across archives, where egs.job.archive.scp generated "
+                        "randomly and examples are combined "
+                        "across all jobs as eg.archive.scp.")
+    parser.add_argument("--random-lang", type=str, action=common_lib.StrToBoolAction,
+                        help="If true, the lang-id in ranges.* selected"
+                        " w.r.t frequency distribution of remaining examples in each language,"
+                        " otherwise it is selected sequentially.",
+                      default=True, choices = ["false", "true"])
+    parser.add_argument("--max-archives", type=int, default=1000,
+                        help="max number of archives used to generate egs.*.scp")
+    parser.add_argument("--seed", type=int, default=1,
+                        help="Seed for random number generator")
+
+    parser.add_argument("--minibatch-size", type=int, default=512,
+                        help="The minibatch size used to generate scp files per job. "
+                             "It should be multiple of actual minibatch size.")
+
+    parser.add_argument("--prefix", type=str, default="",
+                        help="Adds a prefix to the range files. This is used to "
+                        "distinguish between the train and diagnostic files.")
+
+    parser.add_argument("--lang2weight", type=str,
+                        help="lang2weight file contains the weight per language to "
+                           "scale output posterior for that language.(format is: "
+                           "<lang-id> <weight>)")
+# now the positional arguments
+    parser.add_argument("num_langs", type=int,
+                        help="num of languages used in multilingual training setup.")
+    parser.add_argument("egs_scp_lists", type=str,
```

Comment:
it's rather strange, in my opinion, to provide a variable number of arguments as a single string with spaces in it like this.  Look at generate_plots.py, you can use

`nargs='+'`
and it looks like it gives you either None or a list.